### PR TITLE
Feature/particle type

### PR DESCRIPTION
**Pull request by cg-laser**
add more documentation of when to use the particle type, addresses #400

### CODE REVIEW

**Comment by boboeyen** [comment]
Great, thanks for the quick addition. I think It can maybe be even more clearly stipulated that this module is only intended to use for cosmic rays and surface antennas. That would make it easier to understand its use.

**Review by boboeyen — APPROVED** [review]
Great, thanks for the quick addition. I think It can maybe be even more clearly stipulated that this module is only intended to use for cosmic rays and surface antennas. That would make it easier to understand its use.

**Comment by cg-laser** [comment]
but this is not correct. This module can also be used for neutrino signals as well as to reconstruct pulser signals. We used it for example for the ARIANNA SPICE reconstruction. The requirement is that the efield is similar enough at the antennas that are used for reconstruction.

**Comment by boboeyen** [comment]
Ah, then I didn't understand it well enough

File: NuRadioReco/framework/base_station.py [hunk 1]
```diff
@@ -186,8 +186,11 @@ def is_neutrino(self):
 
     def is_cosmic_ray(self):
         if self._particle_type == '':
-            logger.error("Stations particle type has not been set")
-            raise ValueError("Stations particle type has not been set")
+            msg = "Stations particle type has not been set. Please call the module `eventTypeIdentifier.run(event, station, mode='forced', forced_event_type='neutrino'/'cosmic_ray')`."
+            msg += " This flag is used to differentiate between signals that originate from air vs. signals that originate from within the ice which is needed to e.g. determine if refraction "
+            msg += "into the ice needs to be considered."
+            logger.error(msg)
+            raise ValueError(msg)
```
Comment (sjoerd-bouma) [inline NuRadioReco/framework/base_station.py:193]: If we add this to the `is_cosmic_ray` method, shouldn't we maybe add it to the `is_neutrino` method too?
It also is very detailed for an error message (maybe you can copy it over to the docstring, too?), but I guess there's no such thing as a too helpful error message : )

File: NuRadioReco/modules/voltageToEfieldConverter.py [hunk 2]
```diff
@@ -109,6 +109,13 @@ def stacked_lstsq(L, b, rcond=1e-10):
 
 
 class voltageToEfieldConverter:
+    """
+    This module reconstructs the electric field by solving the system of equation that related the incident electric field via the antenna response functions to the measured voltages 
+    (see Eq. 4 of the NuRadioReco paper https://link.springer.com/article/10.1140/epjc/s10052-019-6971-5).
+    The module assumed that the electric field is identical at the antennas/channels that are used for the reconstruction. Furthermore, at least two antennas with
+    orthogonal polarization response are needed to reconstruct the 3dim electric field. 
+    Alternatively, the polarization of the resulting efield could be forced to a single polarization component. In that case, a single antenna is sufficient.
```
Comment (sjoerd-bouma) [inline NuRadioReco/modules/voltageToEfieldConverter.py:117]: Again, very minor, so feel free to ignore. But numpydoc recommends
```
"""
short single line description

(optional) longer description
(...)
""""
```
for docstrings. For a class, I don't actually think it matters at all, but for methods and functions, having long blocks of text without the single line description occasionally renders a bit funny on sphinx.

**Review by sjoerd-bouma — APPROVED** [review]
Thanks Christian for adding some explanation / docstrings here... couple of very small nitpicky points which you are free to ignore. In general, I'm not sure what's good practice with regards to this stuff - it seems unlikely (and also not desirable) that NuRadioMC will ever conform to _all_ of python's / numpy's style guidelines. So let's say I'm happy to be ignored from time to time when it comes to this kind of comments.

**Comment by cg-laser** [comment]
Hey Sjoerd, thanks a lot for your comments. It is very good that you keep educating me/us on how to do better documentation. As you said, I wouldn't be too strict in keeping all standards, it is always a compromise between better documentation and reducing the initial threshold for new people. As any documentation is better than no documentation, we should allow for non-standard doc strings and keep educating people to improve future work.

**Review by sjoerd-bouma — APPROVED** [review]
Thanks!

**Pull request merged**